src/graphql/magentoMapper.js

```diff
@@ -0,0 +1,294 @@
+/*******************************************************************************
+ *
+ *    Copyright 2018 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+'use strict';
+
+const formatDate = require('@adobe/commerce-cif-magento-common/utils').formatDate;
+
+/**
+ * We export a closure function that is initialised with the Openwhisk action parameters required to get
+ * the image URL prefix and the non-variant product attributes.
+ * 
+ * That closure expects that the config argument object has the properties 'imageUrlPrefix'
+ * and 'baseProductAttributes'.
+ * 
+ * @param {*} config 
+ */
+const configureMapper = (config) => {
```

Comment:
Can you maybe refactor this whole thing in a class? And please add documentation.

Reply:
That would indeed be nice but unfortunately the "common" GraphQL library expects a simple object with "dumb" mapper functions,  and doesn't get the object itself so one cannot expect that it would for example call `this.mapper()`. That's why I introduced a closure: we can have state but do not rely on a more traditional "object-oriented construct".